config/filesystems.php

```diff
@@ -59,11 +59,31 @@
             'driver' => 's3',
             'key' => env('AWS_ACCESS_KEY_ID'),
             'secret' => env('AWS_SECRET_ACCESS_KEY'),
-            'region' => env('AWS_DEFAULT_REGION'),
-            'bucket' => env('AWS_BUCKET'),
-            'url' => env('AWS_URL'),
+            'region' => env('AWS_DEFAULT_REGION', 'us-east-1'),
+            'bucket' => env('AWS_BUCKET', get_aws_bucket_from_env()),
+            'url' => env('AWS_URL', get_aws_url_from_env()),
         ],
 
     ],
 
 ];
+
+function get_url_path($url) {
+    $parsed = parse_url($url);
+    return array_key_exists('path', $parsed) ? $parsed['path'] : '';
```

Comment:
Minor nit, this can be:

`$parsed['path'] ?? ''`

Reply:
make sure it's a string